### DISCUSSION TITLE
turbolinks not working with livewire 2

### DISCUSSION THREAD

**Discussion by sxshateri** [livewire/livewire#1946] [1. Help] [answered]
Ok so as per documents for this to work you just have to add it after the livewire script and we will take care of the rest. so i did but nothing changed. the whole page is still loading on every click. so what am i doing wrong?

**Comment by joshhanley** [comment]
Sorry, can you provide some more information? Add what after the livewire script? Which part of the documentation are you referring to? What are you trying to get running?

**Reply by sxshateri** [reply]
> Sorry, can you provide some more information? Add what after the livewire script? Which part of the documentation are you referring to? What are you trying to get running?

sorry i meant to post this for the turbolink repo. posted it here by mistake. not sure if this is the right place to ask help about turbolinks.

**Reply by joshhanley** [reply]
All good! 

As it is here now, I'd say leave it, but maybe change your title to include that it is a turbolinks issue so people know what the issue is about.

**Comment by joshhanley** [comment, marked as answer]
So I just had a quick look at the turbo links adapter docs, and silly question, but have you pulled in turbolinks? 

If not, it's recommended to use NPM on the turbolinks repo, but if you are just testing, add this below your livewire scripts, but above the livewire turbolinks plugin script

```
<script src="https://cdn.jsdelivr.net/npm/turbolinks@5.2.0/dist/turbolinks.min.js"></script>
```

**Reply by sxshateri** [reply]
Thanks for spending time answering my silly questions. in fact i did pull in the turbolinks using NPM but it was not working. then I added the turbolinks using the CDN and it started working. then I remove the CDN script and it stopped working again then i pulled in the turbolinks again using NPM, added it to the app.js again, and compiled the resources and it magically started to work. its exactly what I did the first time but apparently the second time was a charm.

**Reply by joshhanley** [reply]
I don’t think they were silly, I figured you’d already tried to pull in turbolinks (hence my silly question)! 

I’m glad you got it working though. That’s weird about npm. Maybe it was a browser cache issue or something like that, who knows. Do you have versioning turned on your for assets?

**Reply by sxshateri** [reply]
i just started this project today and adding the turbolinks and making sure its working was the first thing I do before setting up anything else such as version controls. if it was on then maybe we could solve this mystery of why it didn't work the first time.

**Reply by joshhanley** [reply]
Ah ok, yeah fair enough
